src/rtutils.c

```diff
@@ -397,7 +402,7 @@ JL_DLLEXPORT void jl_set_nth_field(jl_value_t *v, size_t idx0, jl_value_t *rhs)
 {
     jl_datatype_t *st = (jl_datatype_t*)jl_typeof(v);
     if (!st->name->mutabl)
-        jl_errorf("setfield!: immutable struct of type %s cannot be changed", jl_symbol_name(st->name->name));
+        jl_immutable_field_error(st, (jl_sym_t*)jl_svecref(jl_field_names(st), idx0));
```

Comment:
we might have same issue here with `jl_field_names` for namedtuples. Not sure if it gets hit for namedtuple though.